File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* searchRange(int* nums, int numsSize, int target, int* returnSize) {
     int f=-1,l =-1;
     for (int i=0;i<numsSize;i++){
         if(nums[i] == target){
             if(f==-1){
                f = i;
             }
             l = i;
         }
     }
     int* ans = (int*)malloc(2* sizeof(int));
     ans[0] = f;
     ans[1] = l;
     *returnSize = 2;
     return ans;
}

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        int[] ans ={-1,-1};
        int start = searching(nums,target,true);
        int end = searching(nums,target,false);
        ans[0] = start;
        ans[1] = end;
        return ans; 
    }

    public static int searching(int[] nums,int target,boolean index){
        int ans=-1;
        int start = 0;
        int end = nums.length -1;
        while(start <= end){
            int mid = start + (end-start)/2;
            if(nums[mid]== target){
                ans = mid;
                if(index){
                    end = mid -1;
                }
                else{
                    start = mid+1;
                }
            }
            else if (nums[mid] < target){
                start = mid +1;
            }
            else{
                end = mid -1;
            }
        }
        return ans;
    }
}